### PR TITLE
JavaScript Projec #4: Weather app

### PR DESCRIPTION
**Pull request by mateomh**
This project was created to get the weather information from any city using the openweather API. It has the following features:
- On top of the page you have a text field to put the name of the city and a button to get the information.
- On top of the page there is also a dropdown menu where you select the units that you want to see the temperatures in.
- The weather data is displayed in two different panels, One on the left where the basic information is displayed and one on the right where the other relevant information about the weather is shown.
- Each panel has a random image obtained through the Unsplash API. The left panel will display an image of the requested city, and the right panel will get an image of the current weather.
- If the city is not found it will display error images in the backgrounds and display the errors in the information fields for the city.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,55 @@
...
+- API
+
+## Live Demo
+
+[Live Demo Link](https://raw.githack.com/mateomh/JS-weather-app/weather-app/dist/index.html)
+
+
+### Usage
+
+Everything you need to run the page is inside the **dist** folder of the project.
```
Comment (lucasmazo23) [inline README.md]: [REQUIRED] Please be more specific about what do we need to do with the files in the dist folder

The live version works fine, but when I try to run it locally, it doesn't load the css styles

 
image


image
[source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)

File: src/data.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    temp: data.main.temp,
+    weather: data.weather[0].main,
+    weather_desc: data.weather[0].description,
+    min_temp: data.main.temp_min,
+    max_temp: data.main.temp_max,
+    humidity: data.main.humidity,
+    lat: data.coord.lat,
+    lon: data.coord.lon,
+  };
+}
```
Comment (lucasmazo23) [inline src/data.js]: [REQUIRED] Please try to use ES6 syntax, I'm marking this lines of code, but all the functions in this file could apply the same modifications.
Remember object destructuring, and arrow functions from the previous lesson
image
[source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

File: src/data.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  };
+
+  unsplashApiURL += `?orientation=${orientation}`;
+  unsplashApiURL += `&count=${count}`;
+  unsplashApiURL += `&client_id=${clientID}`;
+  unsplashApiURL += `&query=${type}`;
+
+  const response = await fetch(unsplashApiURL, fetchOptions);
+  const [realData] = await response.json();
+
```
Comment (lucasmazo23) [inline src/data.js]: [REQUIRED] This function needs a try/catch block to handle fetching errors (this is not the only async function without error handling )

image
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

**Review by lucasmazo23 — CHANGES_REQUESTED** [review]
## Changes required 🔧
### You are doing a great job so far, and the project is looking good.

### For the project to get approved kindly implement these changes:

### Required changes:
- [x] [REQUIRED] Please try to use ES6 syntax (Check my comment below for more info) remember object destructuring, and arrow functions from the previous lesson

image
[source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

- [x] [REQUIRED] Some functions need try/catch blocks to handle fetching errors (check the comment below for more info)

image
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

- [x] [REQUIRED] Please be more specific about what do we need to do with the files in the dist folder

image
[source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)


### Happy coding! 💻
## Lucas Mazo

**Review by bellom — APPROVED** [review]
## Hi @mateomh  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**